src/finchlite/finch_assembly/cfg.py

```diff
@@ -0,0 +1,282 @@
+                | SetAttr()
+                | Call()
+                | Load()
+                | Store()
+                | Length()
+                | Slot()
+                | Stack()
+                | Assign()
+            ):
+                self.current_cfg.current_block.add_statement(node)
+            case Block(bodies):
+                for body in bodies:
+                    self(body, break_block)
+            case If(cond, body):
+                before_block = self.current_cfg.current_block
+
+                if_block = self.current_cfg.new_block()
+                after_block = self.current_cfg.new_block()
+
+                before_block.add_successor(if_block)
+                before_block.add_successor(after_block)
+
+                if_block.add_statement(cond)
+                self.current_cfg.current_block = if_block
+                self(body, break_block)
+
+                self.current_cfg.current_block.add_successor(after_block)
+                self.current_cfg.current_block = after_block
+            case IfElse(cond, body, else_body):
+                before_block = self.current_cfg.current_block
+
+                if_block = self.current_cfg.new_block()
+                else_block = self.current_cfg.new_block()
+                after_block = self.current_cfg.new_block()
+
+                before_block.add_successor(if_block)
+                before_block.add_successor(else_block)
+
+                if_block.add_statement(cond)
+                self.current_cfg.current_block = if_block
+                self(body, break_block)
+                self.current_cfg.current_block.add_successor(after_block)
+
+                self.current_cfg.current_block = else_block
+                self(else_body, break_block)
+                self.current_cfg.current_block.add_successor(after_block)
+
+                self.current_cfg.current_block = after_block
+            case WhileLoop(cond, body):
+                before_block = self.current_cfg.current_block
+
+                body_block = self.current_cfg.new_block()
+                after_block = self.current_cfg.new_block()
+
+                before_block.add_successor(body_block)
+                before_block.add_successor(after_block)
+
+                body_block.add_statement(cond)
+                self.current_cfg.current_block = body_block
+                self(body, after_block)
+
+                self.current_cfg.current_block.add_successor(before_block)
+                self.current_cfg.current_block = after_block
+            case ForLoop(var, _start, _var, body):
+                before_block = self.current_cfg.current_block
+
+                body_block = self.current_cfg.new_block()
+                after_block = self.current_cfg.new_block()
+
+                before_block.add_successor(body_block)
+                before_block.add_successor(after_block)
+
+                # TODO: figure out a RIGHT way to represent 'ForLoop' initialization
```

Comment:
I think we should introduce a new variable, so that
```
for i = start:stop
    body
```
becomes
```
j = start
while j <= end
    i = j
    body
    j = j + 1
```
Then we can handle it like the while case.